src/Core/Workspace/Workspace.cs

```diff
@@ -382,6 +422,8 @@ private void LoadReferencedPackages(Action<string> statusCallback = null)
         /// <inheritdoc/>
         public void AddProject(string projectFile)
```

Comment:
It looks like both this and `Reload` are blocking due to calls to `WaitForInitialization`; are they called from async methods further up?

Reply:
In IQ#, those methods are currently called only from the `Run()` method of magic command implementations (`%workspace reload` and `%project`), and yes, those are being called in an async call via `AbstractMagic.SafeExecute()`.

I take your point though that both `Reload()` and `AddProject()` could be long-running calls and should be async APIs themselves. Actually, `Reload()` was already long-running since it is synchronously waiting for compilation and package downloads. So, while maybe it should be async, I would like to avoid introducing that API change at the moment, if possible.

And upon further thought, there is no reason for calling `WaitForInitialization()` from inside `AddProject()`, since that method is not actually doing any compilation (and the magic command entry point will anyway not be invoked until the initialization has completed). I'll remove that one.

Reply:
Makes sense!